common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java

```diff
@@ -379,10 +379,19 @@ public boolean isDisabled() {
         }
     }
 
-    public void flush() {
-        // TODO: flush() likely needs some form of synchronization mechanism to notify the caller
-        // when the flush is complete. For example, a promise, a callback, etc.
-        flushInternal(true);
+    /**
+     * Initiates a flush of data to the collectors.
```

Comment:
Minor, but it'd probably be good to explicitly note that this is a blocking call.